packages/in_app_purchase/example/ios/in_app_purchase_pluginTests/in_app_purchase_pluginTests.m

```diff
@@ -16,21 +16,47 @@ - (void)tearDown {
 }
 
 - (void)testInvalidMethodCall {
+  XCTestExpectation* expectation =
+      [self expectationWithDescription:@"expect result to be not implemented"];
   FlutterMethodCall* call = [FlutterMethodCall methodCallWithMethodName:@"invalid" arguments:NULL];
+  __block id result;
   [plugin handleMethodCall:call
-                    result:^(id result) {
-                      XCTAssert(result == FlutterMethodNotImplemented);
+                    result:^(id r) {
+                      [expectation fulfill];
+                      result = r;
                     }];
+  [self waitForExpectations:@[ expectation ] timeout:5];
+  XCTAssertEqual(result, FlutterMethodNotImplemented);
 }
 
 - (void)testCanMakePayments {
+  XCTestExpectation* expectation = [self expectationWithDescription:@"expect result to be YES"];
   FlutterMethodCall* call =
       [FlutterMethodCall methodCallWithMethodName:@"-[SKPaymentQueue canMakePayments:]"
                                         arguments:NULL];
+  __block id result;
   [plugin handleMethodCall:call
-                    result:^(id result) {
-                      XCTAssert(result == [NSNumber numberWithBool:YES]);
+                    result:^(id r) {
+                      [expectation fulfill];
+                      result = r;
                     }];
+  [self waitForExpectations:@[ expectation ] timeout:5];
+  XCTAssertEqual(result, [NSNumber numberWithBool:YES]);
+}
+
+- (void)testGetProductListEmpty {
```

Comment:
Is there any way we can unit test an actually successful response here, or is it not mockable?